datastore/longhorn.go

```diff
@@ -494,6 +504,49 @@ func (s *DataStore) ValidateV2DataEngineEnabled(dataEngineEnabled bool) (ims []*
 	return
 }
 
+func (s *DataStore) ValidateCPUMask(value string) error {
+	// CPU mask must start with 0x
+	cpuMaskRegex := regexp.MustCompile(`^0x[1-9a-fA-F][0-9a-fA-F]*$`)
+	if !cpuMaskRegex.MatchString(value) {
+		return fmt.Errorf("invalid CPU mask: %s", value)
+	}
+
+	maskValue, err := strconv.ParseUint(value[2:], 16, 64) // skip 0x prefix
+	if err != nil {
+		return fmt.Errorf("failed to parse CPU mask: %s", value)
+	}
+
+	// Validate the mask value is not larger than the number of available CPUs
+	numCPUs := runtime.NumCPU()
+	maxCPUMaskValue := (1 << numCPUs) - 1
+	if maskValue > uint64(maxCPUMaskValue) {
+		return fmt.Errorf("CPU mask exceeds the maximum allowed value for the current system: %s", value)
```

Comment:
nit: add the max value info in the error message.

Reply:
Updated. Thank you.